rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -0,0 +1,267 @@
+                    event.counterparty = CPT_AAVE_V2
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_withdrawal(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 withdrawal event"""
+        user = hex_or_bytes_to_address(tx_log.topics[2])
+        to = hex_or_bytes_to_address(tx_log.topics[3])
+        if self.base_tools.is_tracked(user) is False and self.base_tools.is_tracked(to) is False:  # noqa: E501
+            return
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data),
+            asset=token,
+        )
+        notes = f'Withdraw {amount} {token.symbol} from AAVE v2'
+        if to != user:
+            notes += f' to {to}'
+        for event in decoded_events:
+            if (
+                event.balance.amount == amount and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                if (
+                    event.location_label == to and
+                    event.event_type == HistoryEventType.RECEIVE
+                ):
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.location_label = user
+                    event.notes = notes
+                else:
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    resolved_asset = event.asset.resolve_to_asset_with_symbol()
+                    event.notes = f'Return {amount} {resolved_asset.symbol} to AAVE v2'  # noqa: E501
+                # Set counterparty for both events
+                event.counterparty = CPT_AAVE_V2
+
+    def _decode_borrow(
+            self,
+            token: 'EvmToken',
+            tx_log: EvmTxReceiptLog,
+            decoded_events: list[HistoryBaseEntry],
+    ) -> None:
+        """Decode aave v2 borrow event"""
+        on_behalf_of = hex_or_bytes_to_address(tx_log.topics[2])
+        user = hex_or_bytes_to_address(tx_log.data[:32])
+        if self.base_tools.is_tracked(user) is False and self.base_tools.is_tracked(on_behalf_of) is False:  # noqa: E501
+            return
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(tx_log.data[32:64]),
+            asset=token,
+        )
+        rate_mode = hex_or_bytes_to_int(tx_log.data[64:96])
+        # Aave uses ray math https://docs.aave.com/developers/v/1.0/developing-on-aave/important-considerations#ray-math  # noqa: E501
+        # To get the rate we have to divide the value by 1e27. And then multiply by 100 to get the percentage.  # noqa: E501
+        rate = hex_or_bytes_to_int(tx_log.data[96:128]) / FVal(1e25)
```

Comment:
Use the already existing RAY constant. It's currently at modules/makerdao/constants.py.

Probably should move to a more central place.